### PR TITLE
add boot partition layout section for raspberry pi

### PR DESCRIPTION
**Pull request by wfeldt**
After some talk with Alex Graf, here's what I understood.

### CODE REVIEW

File: doc/boot-partition.md [hunk 1]
```diff
@@ -250,7 +250,29 @@ AI Ihno:
...
 
+- see also fate #323484, bsc #1041475
+
+- RPi boots from sdcard, must have a msdos partition table
+
+- first, a firmware blob is read; this has to be on a dedicated
+  vfat partition that must not be deleted, possibly this partition must be the
+  first partition (check!)
+
+- this firmware then loads a bootloader (u-boot)
```
Comment (agraf) [inline doc/boot-partition.md]: which also resides on the firmware partition

File: doc/boot-partition.md [hunk 2]
```diff
@@ -250,7 +250,29 @@ AI Ihno:
...
+
+- u-boot then does the typical UEFI boot proceedure (loading grub2, etc)
```
Comment (agraf) [inline doc/boot-partition.md]: ... but requires the ESP to be marked as active

File: doc/boot-partition.md [hunk 3]
```diff
@@ -250,7 +250,29 @@ AI Ihno:
...
+
+- this special RPi partition contains some firmware blob (file name?),
+  u-boot, and a lot of RPi config files
+
+- installation can be via network or from usb-stick, selectable in u-boot (via menu)
```
Comment (agraf) [inline doc/boot-partition.md]: U-Boot doesn't do menus. The menu would come from an "installer" grub on the firmware partition.

File: doc/boot-partition.md [hunk 4]
```diff
@@ -250,7 +250,29 @@ AI Ihno:
 
 - provide a script to detect DASD type (FBA, DIAG...), is it allowed to create a partition table?
 
+## aarch64 (RPi 3)
 
+- see also fate #323484, bsc #1041475
+
+- RPi boots from sdcard, must have a msdos partition table
```
Comment (mvidner) [inline doc/boot-partition.md]: let's spell out the name in full at least once, in case we need to grep for it
> Raspberry Pi

Comment (mvidner) [inline doc/boot-partition.md]: I have looked for the [upstream docs](https://github.com/raspberrypi/documentation/tree/master/hardware/raspberrypi/bootmodes), found it, and it [says:](https://github.com/raspberrypi/documentation/blob/master/hardware/raspberrypi/bootmodes/bootflow.md)

> The boot ROM also now supports GUID partitioning and has been tested with hard drives partitioned using Mac, Windows, and Linux.

File: doc/boot-partition.md [hunk 5]
```diff
@@ -250,7 +250,29 @@ AI Ihno:
...
+
+- first, a firmware blob is read; this has to be on a dedicated
+  vfat partition that must not be deleted, possibly this partition must be the
+  first partition (check!)
```
Comment (mvidner) [inline doc/boot-partition.md]: > It is no longer necessary for the first partition to be the FAT partition, as the MSD boot will continue to search for a FAT partition beyond the first one.

:)

File: doc/boot-partition.md (hunk 1 above)
Comment (mvidner) [inline doc/boot-partition.md]: I did not know what that was, maybe add a link: http://elinux.org/RPi_U-Boot

File: doc/boot-partition.md (hunk 4 above)
Comment (agraf) [inline doc/boot-partition.md]: That sounds awesome. Certainly needs to be verified though :).

File: doc/boot-partition.md (hunk 5 above)
Comment (wfeldt) [inline doc/boot-partition.md]: Do you have an external reference for this?

Comment (mvidner) [inline doc/boot-partition.md]: Ah, sorry, apparently MSD means a (USB) Mass Storage Device, not the SD card.

File: doc/boot-partition.md [hunk 6]
```diff
@@ -270,7 +297,8 @@ AI Ihno:
 - [PowerLinux Boot howto](https://www.ibm.com/developerworks/community/wikis/home?lang=en#!/wiki/W51a7ffcf4dfd_4b40_9d82_446ebc23c550/page/PowerLinux%20Boot%20howto)
 - [SLOF - Slimline Open Firmware](https://github.com/aik/SLOF/blob/master/slof/fs/packages/disk-label.fs)
 - [GRUB Documentation](https://www.gnu.org/software/grub/grub-documentation.html)
-- [GRUB wikipedia] (https://de.wikipedia.org/wiki/Grand_Unified_Bootloader)
+- [GRUB wikipedia](https://de.wikipedia.org/wiki/Grand_Unified_Bootloader)
+- [U-Boot](http://elinux.org/RPi_U-Boot)
```
Comment (mvidner) [inline doc/boot-partition.md]: Please add
```
[Raspberry Pi boot modes](https://github.com/raspberrypi/documentation/tree/master/hardware/raspberrypi/bootmodes)
```

Comment (wfeldt) [inline doc/boot-partition.md]: done

**Pull request merged**